api4/channel.go

```diff
@@ -690,8 +690,13 @@ func getPinnedPosts(c *Context, w http.ResponseWriter, r *http.Request) {
 }
 
 func getAllChannels(c *Context, w http.ResponseWriter, r *http.Request) {
-	if !c.App.SessionHasPermissionTo(*c.App.Session(), model.PERMISSION_MANAGE_SYSTEM) {
-		c.SetPermissionError(model.PERMISSION_MANAGE_SYSTEM)
+	permissions := []*model.Permission{
+		model.PERMISSION_READ_SYSCONSOLE_USERMANAGEMENT,
+		model.PERMISSION_READ_SYSCONSOLE_USERMANAGEMENT_GROUPS,
+		model.PERMISSION_READ_SYSCONSOLE_USERMANAGEMENT_CHANNELS,
```

Comment:
is the first one `PERMISSION_READ_SYSCONSOLE_USERMANAGEMENT` needed? 
Maybe im misunderstanding but shouldnt a role that has `PERMISSION_READ_SYSCONSOLE_USERMANAGEMENT` automatically have both `PERMISSION_READ_SYSCONSOLE_USERMANAGEMENT_GROUPS` and `PERMISSION_READ_SYSCONSOLE_USERMANAGEMENT_CHANNELS`? so really only those two permissions need to be checked?

Reply:
A previous iteration of this had a parent permission for that section which one could optionally have to encompassing being granted all of the sub-sections. I have since remove the parent section for logical simplicity.